src/visualizations/config/adapters/dhis_highcharts/customSVGOptions/singleValue/renderer/renderSingleValueSVG.js

```diff
@@ -0,0 +1,53 @@
+import { addIconElement } from './addIconElement.js'
+import { checkIfFitsWithinContainer } from './checkIfFitsWithinContainer.js'
+import { getAvailableSpace } from './getAvailableSpace.js'
+import { positionElements } from './positionElements.js'
+import { DynamicStyles } from './styles.js'
+
+export function renderSingleValueSVG() {
+    const { formattedValue, icon, subText, fontColor } =
+        this.userOptions.customSVGOptions
+    const dynamicStyles = new DynamicStyles(this.userOptions?.isPdfExport)
+    const valueElement = this.renderer
+        .text(formattedValue)
+        .css({ color: fontColor, visibility: 'hidden' })
+        .add()
+    const subTextElement = subText
+        ? this.renderer
+              .text(subText)
+              .css({ color: fontColor, visibility: 'hidden' })
+              .add()
+        : null
+    const iconElement = icon ? addIconElement.call(this, icon, fontColor) : null
+
+    let fitsWithinContainer = false
+    let styles = {}
+
+    while (!fitsWithinContainer && dynamicStyles.hasNext()) {
+        styles = dynamicStyles.next()
+
+        valueElement.css(styles.value)
+        subTextElement?.css(styles.subText)
+
+        fitsWithinContainer = checkIfFitsWithinContainer(
+            getAvailableSpace.call(this, styles.spacing.valueTop),
+            valueElement,
+            subTextElement,
+            icon,
+            subText,
+            styles.spacing
+        )
+    }
```

Comment:
This part of the code is probably the most essential part of this PR. The mechanism is simple:
1. We start by rendering the required elements in the biggest size
2. We then check if it fits
3. And repeat the process until it does

So the scaling is not dynamic, it happens in steps. The reason for this is that this way it is more likely that multiple dashboard SV items will end up having the same font size.